index.html

```diff
@@ -296,141 +292,176 @@
+  // Handles exceptions, also handles special weekend schedules
+  function getDaySchedule(dateObj, exceptions) {
     var day = dayOfWeekToString(dateObj.getDay());
-    // Handles exceptions, also handles special weekend schedules
-    for (var exception = 0; exception < days[1].length; exception++) {
-        if (days[1][exception]["day"] == dateObj.getDate() &&
-            days[1][exception]["month"] == (dateObj.getMonth() + 1) &&
-            days[1][exception]["year"] == dateObj.getFullYear()) {
-            day = days[1][exception]["schedule"];
+    for (var exception = 0; exception < exceptions[1].length; exception++) {
+        if (exceptions[1][exception]["day"] == dateObj.getDate() &&
+            exceptions[1][exception]["month"] == (dateObj.getMonth() + 1) &&
+            exceptions[1][exception]["year"] == dateObj.getFullYear()) {
+            day = exceptions[1][exception]["schedule"];
         }
     }
-    var currentSlot;
-
-    // Will not get called if there is no school
-    for (currentSlot = 0; currentSlot < days[0][day][school].length; currentSlot++) {
-        switch(days[0][day][school][currentSlot]["period"]) { // Falling through intended
-            case "0": // 0 Period
-            case "A": // A period
-            case "B": // B period
-            case "C": // C period
-            case "D": // D period
-            case "E": // E period
-            case "F": // F period
-            case "G": // G period
-            case "H": // H period
-                for (var k = 0; k < schedule["classes"].length; k++) {
-                    if (schedule["classes"][k]["period"] == days[0][day][school][currentSlot]["period"]) {
-                        if (schedule["classes"][k]["name"] != "Free Period") {
-                            var teacherName = schedule["classes"][k]["teacher"]; // Teacher's fullname
-                            var teacherLast = teacherName.split(" ")[1]; // Teacher's last name
-                            scheduleObj = { name: schedule["classes"][k]["name"],
-                                teacher: teacherName,
-                                teacherLastName: teacherLast,
-                                room: schedule["classes"][k]["room"],
-                                period: days[0][day][school][currentSlot]["period"],
-                                time: ""}
+    return day
+  }
 
-                        } else {
-                            scheduleObj = {name: "Free Period",
-                                teacher: "",
-                                teacherLastName: "",
-                                room: "",
-                                period: days[0][day][school][currentSlot]["period"],
-                                time: ""};
-                        }
-                    }
-                }
-                break;
+  // Generate basic name, teacher, room, time, and period info for a class
+  function renderClassBase(schedule, school, day, currentSlot) {
```

Comment:
Since this returns a class entry object, perhaps call this createClassEntry?

When you return a new object, it's good to call the function createXXXX.